src/lib.rs

```diff
@@ -45,26 +45,19 @@ pub async fn run_trin(
     let (node_data_dir, private_key) =
         configure_node_data_dir(trin_data_dir, trin_config.private_key)?;
 
-    let portalnet_config = PortalnetConfig {
-        external_addr: trin_config.external_addr,
-        private_key,
-        listen_port: trin_config.discovery_port,
-        no_stun: trin_config.no_stun,
-        bootnodes: trin_config.bootnodes.clone(),
-        enr_file_location: Some(node_data_dir.clone()),
-        ..Default::default()
-    };
+    // Initialize prometheus metrics
+    if let Some(addr) = trin_config.enable_metrics_with_url {
+        prometheus_exporter::start(addr)?;
+    }
```

Comment:
The move of the exporter up here used to be required because we were passing it into the config, but we're not doing that anymore. So I think the code move is unnecessary, right?

If it is necessary, a comment explaining why is warranted. If not, *nit* to reduce code turnover a bit by moving the block back into place.